### PR TITLE
[improve](fe): use `Pair.of` to replace `new Pair<>()`

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem summary

use `Pair.of` to replace `new Pair<>()`

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/common/Pair.java [hunk 1]
```diff
@@ -37,41 +38,31 @@
...
-    public Pair(F first, S second) {
+    private Pair(F first, S second) {
         this.first = first;
         this.second = second;
     }
 
-    public static <F, S> Pair<F, S> create(F first, S second) {
-        return new Pair<F, S>(first, second);
+    public static <F, S> Pair<F, S> of(F first, S second) {
+        return Pair.of(first, second);
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/common/Pair.java:47]: endless loop？

Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/common/Pair.java]: fixed

**Comment by stalary** [comment]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**